index.js

```diff
@@ -88,17 +88,29 @@ function doInject (dispatchFunc, options, callback) {
 
   const server = options.server || {}
 
+  const makeRequest = (req, res) => {
+    req.once('error', function (err) {
+      if (this.destroyed) res.destroy(err)
+    })
+
+    req.once('close', function () {
+      if (this.destroyed && !this._error) res.destroy()
+    })
+
+    return req.prepare(() => dispatchFunc.call(server, req, res))
+  }
```

Comment:
Can you move `makeRequest` one level up? There is no need to allocate a new one for every inject.